packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -139,26 +143,31 @@ export class AzureBlobStoragePublish implements PublisherBase {
           `${entityRootDir}/${relativeFilePath}`,
         ); // Azure Blob Storage Container file relative path
 
-        return limiter(async () => {
-          await uploadPromises.push(
-            this.storageClient
-              .getContainerClient(this.containerName)
-              .getBlockBlobClient(destination)
-              .uploadFile(filePath),
-          );
+        return limiter(() => {
+          return this.storageClient
+            .getContainerClient(this.containerName)
+            .getBlockBlobClient(destination)
+            .uploadFile(filePath);
         });
       });
 
-      await Promise.all(promises).then(() => {
+      let responses: BlobUploadCommonResponse[] = [];
+
+      responses = await Promise.all(promises);
+
+      const failed = responses.filter(r => r?._response?.status >= 400);
+      if (failed.length === 0) {
         this.logger.info(
           `Successfully uploaded all the generated files for Entity ${entity.metadata.name}. Total number of files: ${allFilesToUpload.length}`,
         );
-      });
-      return;
+      } else {
+        const errorMessage = `Unable to upload ${failed.length} file(s) to Azure Blob Storage.`;
+        this.logger.error(errorMessage);
```

Comment:
We should throw error in this block - or else the promise will be resolved.

Reply:
Done in https://github.com/backstage/backstage/pull/4539/commits/02c5ce979c49aac15b6e4dde159905598d75e79a